src/engine/input/key_mapping/InputEventItem.cs

```diff
@@ -0,0 +1,210 @@
+using Godot;
+
+/// <summary>
+///   Defines one specific event associated to an <see cref="InputActionItem">InputActionItem</see>.
+/// </summary>
+public class InputEventItem : Node
+{
+    [Export]
+    public NodePath ButtonPath;
+
+    [Export]
+    public NodePath XButtonPath;
+
+    private Button button;
+    private Button xbutton;
+    private bool wasPressingButton;
+
+    /// <summary>
+    ///   If it is currently awaiting the user to press a button
+    /// </summary>
+    public bool WaitingForInput { get; private set; }
+
+    /// <summary>
+    ///   The currently assigned godot input event.
+    ///   null if this event was just created
+    /// </summary>
+    public ThriveInputEventWithModifiers AssociatedEvent { get; set; }
+
+    /// <summary>
+    ///   The action this event is associated with
+    /// </summary>
+    internal InputActionItem AssociatedAction { get; set; }
+
+    /// <summary>
+    ///   If the event was just created and has never been assigned with a value
+    /// </summary>
+    internal bool JustAdded { get; set; }
+
+    /// <summary>
+    ///   Deleted this event from the associated action and updated the godot InputMap
+    /// </summary>
+    public void Delete()
+    {
+        AssociatedAction.Inputs.Remove(this);
+        InputGroupList.Instance.ControlsChanged();
```

Comment:
Instead of accessing a global, would it work if that was passed in as a property through the node tree until it reaches here?

Reply:
I honestly don't get what you mean...

Reply:
I mean that, you create the tree of objects starting from the input group list, which in turn ends up creating this.
Global use should be avoided.
So I suggest you do something like:
input group list -> creates some object and passes itself as reference -> this in turn creates then InputEventItem and passes the reference to input group list

That way there isn't a need to use a global here. As I said, if feasible in cases like this instead of using a global the object should be propagated through to where it is needed.

Reply:
with my latest commits I have made many things static. Did this resolve itself or should I still change this?

Reply:
See my latest comment: https://github.com/Revolutionary-Games/Thrive/pull/1772#discussion_r516594823
It also applies here